### ISSUE TITLE
Change Emulator Language Interface

### ISSUE THREAD

**Issue by RunisDauphin** [visualboyadvance-m/visualboyadvance-m#413]
### Expected behavior
Have an action (like previous VBA-M versions and other VBA branches allow) to set language of the emulator interface into another one. Prevously this was achieved with plugins named vba_???.dll. (? was one letter, and every language had its code) and the option Language -> UI language.

### Actual behavior
There is no posibility (at least that I am aware of) to set any other language, and I have zero clue on how to use any of those locale.mo files.

### Steps to reproduce the behavior
If better, steps to get out of this behaviour.

### Crash bugs
None

If this is a crash bug, and you have the time and inclination, please provide a
symblic backtrace.

Instructions for doing so are
[here](https://github.com/visualboyadvance-m/visualboyadvance-m/blob/master/README.md#reporting-crash-bugs).

### ROM details

Which games are affected by the issue (please state if GBA or GB game): Affects the emulator interface, not games.

Which games, if any, are NOT affected by the issue (please state if GBA or GB game): Affects the emulator interface, not games.

Include screenshots of ROM Information (File > ROM Information): No game is affected by this.

If you are using a BIOS file, please say which one.
The BIOS does not intervene in this process.

### Option details

Video driver (simple, opengl): Irrelevant.

Sound driver (directsound, xaudio, openal, SDL): Irrelevant.

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.): Windows, every build since 2.0.0

If using Linux, specify if using xorg or Wayland:

Version of code (pre-built binary version, or commit ref, or just "master"): Any release version and developpement version since before 2.0.0.

### Hardware details:

If your joystick or gamepad isn't working in Linux, please try to troubleshoot
using the information on this page:
https://wiki.archlinux.org/index.php/Gamepad **BEFORE** filing an issue. We use
SDL for joystick support, any problems will generally be an issue with SDL and
**NOT** the emulator.

If relevant, the joystick you are using and the drivers you are using for it:
I do not think it is relevant, but I use both keyboard or Gamecube controller through the official Wii U GCN USB adapter and these drivers, alongside Zadig and vJoy in their latest version:
http://m4sv.com/page/wii-u-gcn-usb-driver. These drivers are free and have no relation to Nintendo. Works in almost every emulator I tried and some games are compatible too.

**Comment by ZachBacon** [comment]
Currently language switching is achieved by switching your OS language. However I'm investigating a solution that should allow dynamic language switching. But it's not high priority just yet.

**Comment by RunisDauphin** [comment]
OK then, I will stick with what I have. Could you write it on the readme when that is operative again?

**Comment by RunisDauphin** [comment]
If someone wants them, I found some time ago some files to translate VBA onto other languages (although they mysteriously suppressed the Link option on some versions).
[Localisation Files.zip](https://github.com/visualboyadvance-m/visualboyadvance-m/files/3101682/Localisation.Files.zip)

vba_esv.dll translates into Spanish, and vba_fra.dll translates into French. As far as I know, they are compatible with VBA, VBA Link (even that weird "Hack" with e-Reader support compiled by Javier Himura), VBA last SVN version, VBA-RR and first versions of VBA-M (maybe up to r1500?).

vba_esv.dll was obtained from some youtube video I ignore because that was like 3 years ago.
vba_fra.dll was obtained from a post on gbatemp.net where they distributed an old version of VBA I did not want but had this file I did want and I found very convenient.

Hope this helped, and if someone searched for these files, here you can find them (and maybe they can help a bit to developpe the new ones).

Thanks for your attention, M. Bacon and Kitover.

**Comment by ZachBacon** [comment]
That is unnecessary, we have a transifex account that has ongoing translations that people contribute to that we pull from occassionally.

**Comment by RunisDauphin** [comment]
Yeah, I saw them (and I have them, because they are included with every package of yours), but some people may find them useful if they still stick with the old versions I wrote there.

Thanks again, M. Bacon.

**Comment by ZachBacon** [comment]
We won't support old versions. So no point

**Comment by mandrean** [comment]
Any news on this in 2024? Either through GUI or config changes

I would much prefer English over the awkward Swedish ones.

**Comment by mandrean** [comment]
For anyone dropping by here in 2024, at least on macOS this can be done by setting a manual localization override for the vba-m app under `System Settings -> Language & region`

https://support.apple.com/guide/mac-help/change-the-system-language-mh26684/mac

On my Swedish macOS it would look like this:
![image](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/4631744/4b4980ee-a59a-4e11-b60e-a562560c0786)

**Comment by ZachBacon** [comment]
@rkitover has it planned.
